File: 1009.complement-of-base-10-integer.0.js
/*
 * @lc app=leetcode id=1009 lang=javascript
 *
 * [1009] Complement of Base 10 Integer
 */
/**
 * @param {number} N
 * @return {number}
 */
const bitwiseComplement = function(N) {
    if (N == 0) return 1
    let i = 1
    while (i < N) {
      i = i << 1
    }
    return ~N & (i - 1)
}

File: 1009.complement-of-base-10-integer.0.go
/*
 * @lc app=leetcode id=1009 lang=golang
 *
 * [1009] Complement of Base 10 Integer
 */
func bitwiseComplement(N int) int {
    if N == 0 {
        return 1
    }
    i := 1
    for i < N {
        i = i << 1
    }

    return ^N & (i- 1)
}
